packages/base/src/panelview/components/sources.tsx

```diff
@@ -0,0 +1,338 @@
+    setModel(widget?.context.model);
+  });
+
+  return (
+    <div id="jp-gis-sources">
+      {sourceIds.map(sourceId => {
+        return (
+          <SourceComponent
+            key={`source-${sourceId}`}
+            gisModel={model}
+            sourceId={sourceId}
+            onClick={onItemClick}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+/**
+ * Properties of the source component.
+ */
+interface ISourceProps {
+  gisModel: IJupyterGISModel | undefined;
+  sourceId: string;
+  onClick: ({ type, item, nodeId }: SourcesPanel.IClickHandlerParams) => void;
+}
+
+function isSelected(sourceId: string, model: IJupyterGISModel | undefined) {
+  return (
+    (model?.localState?.selected?.value &&
+      Object.keys(model?.localState?.selected?.value).includes(sourceId)) ||
+    false
+  );
+}
+
+/**
+ * The component to display a single source.
+ */
+function SourceComponent(props: ISourceProps): JSX.Element {
+  const { sourceId, gisModel, onClick } = props;
+  const source = gisModel?.getSource(sourceId);
+  if (source === undefined) {
+    return <></>;
+  }
+
+  const [id, setId] = useState('');
+  const [selected, setSelected] = useState<boolean>(
+    // TODO Support multi-selection as `model?.jGISModel?.localState?.selected.value` does
+    isSelected(sourceId, gisModel)
+  );
+  const [unused, setUnused] = useState<boolean>(false);
+  const name = source.name;
+
+  useEffect(() => {
+    setId(DOMUtils.createDomID());
+  }, []);
+
+  /**
+   * Check if the source is used by a layer.
+   */
+  useEffect(() => {
+    const checkUsage = () => {
+      if (!(gisModel?.getLayersBySource(sourceId).length ?? true)) {
+        setUnused(true);
+      }
+    };
+
+    gisModel?.sharedLayersChanged.connect(checkUsage);
+    checkUsage();
+
+    return () => {
+      gisModel?.sharedLayersChanged.disconnect(checkUsage);
+    };
+  }, [gisModel]);
+
+  /**
+   * Listen to the changes on the current source.
+   */
+  useEffect(() => {
+    const onClientSharedStateChanged = (
+      sender: IJupyterGISModel,
+      clients: Map<number, IJupyterGISClientState>
+    ) => {
+      // TODO Support follow mode and remoteUser state
+      setSelected(isSelected(sourceId, gisModel));
+    };
+    gisModel?.clientStateChanged.connect(onClientSharedStateChanged);
+
+    return () => {
+      gisModel?.clientStateChanged.disconnect(onClientSharedStateChanged);
+    };
+  }, [gisModel]);
+
+  const setSelection = (event: MouseEvent<HTMLElement>) => {
+    const childId = event.currentTarget.children.namedItem(id)?.id;
+    onClick({ type: 'source', item: sourceId, nodeId: childId, event });
+  };
+
+  let mainClasses = SOURCE_CLASS;
+  if (selected) {
+    mainClasses = mainClasses.concat(' jp-mod-selected');
+  }
+  if (unused) {
+    mainClasses = mainClasses.concat(` ${SOURCE_UNUSED}`);
+  }
+  return (
+    <div className={mainClasses}>
+      <div
+        className={SOURCE_TITLE_CLASS}
```

Comment:
I added a register in https://github.com/QuantStack/jupytergis/pull/60/commits/763115c00ca84482b29dfcaa61c16e2708b35c82.
There are registered icons for the panels and the commands.
I had to move the `commandIDs` in a separate file to avoid circular import.